data-formats/src/ownershipvoucher.rs

```diff
@@ -450,7 +473,9 @@ impl Serializable for OwnershipVoucherHeader {
         let cached_guid = contents.get(OwnershipVoucherHeaderIndex::Guid as usize)?;
         let cached_rendezvous_info =
             contents.get(OwnershipVoucherHeaderIndex::RendezvousInfo as usize)?;
-        let cached_device_info = contents.get(OwnershipVoucherHeaderIndex::DeviceInfo as usize)?;
+        let cached_device_info: String =
+            contents.get(OwnershipVoucherHeaderIndex::DeviceInfo as usize)?;
+        if !check_device_info(&cached_device_info).unwrap() { /* return an error? */ };
```

Comment:
You could use `check_device_info(&cached_device_info)?;`, and then the error that that returns will just be propagated up.

Reply:
Fixed.

I wanted to use that but the compiler was complaining, that's why I changed it to that `if` statement. Now I put it back, it's compiling so probably it was an error on my early implementation.